docs/index.rst

```diff
@@ -1470,42 +1470,69 @@ Use with tarfile module
 
     Python's `tarfile <https://docs.python.org/3/library/tarfile.html>`_ module supports arbitrary compression algorithms by providing a file object.
 
-    This code encapsulates a ``ZstdTarFile`` class using :py:class:`ZstdFile`, it can be used like `tarfile.TarFile <https://docs.python.org/3/library/tarfile.html#tarfile.TarFile>`_ class:
-
     .. sourcecode:: python
 
         import tarfile
 
-        # when using read mode (decompression), the level_or_option parameter
-        # can only be a dict object, that represents decompression option. It
-        # doesn't support int type compression level in this case.
+        # compression
+        with ZstdFile('archive.tar.zst', mode='w') as _fileobj, tarfile.open(fileobj=_fileobj, mode='w') as tar:
+            # do something
+
+        # decompression
+        with ZstdFile('archive.tar.zst', mode='r') as _fileobj, tarfile.open(fileobj=_fileobj) as tar:
+            # do something
+
+    Alternatively, it is possible to extend the ``Tarfile`` class, except that it supports decompressing ``.tar.zst`` file automatically, as well as adding the following modes: ``r:zst``, ``w:zst`` and ``x:zst``.
```

Comment:
IMO «except» reads strange here. Maybe you meant: extend the TarFile class so that it supports etc ?